rules/no-array-push-push.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+const {hasSideEffect, isCommaToken, isOpeningParenToken, isSemicolonToken} = require('eslint-utils');
+const getDocumentationUrl = require('./utils/get-documentation-url');
+const methodSelector = require('./utils/method-selector');
+
+const ERROR = 'error';
+const SUGGESTION = 'suggestion';
+const messages = {
+	[ERROR]: 'Do not call `Array#push()` multiple times.',
+	[SUGGESTION]: 'Merge with previous one.'
+};
+
+const arrayPushExpressionStatement = [
+	'ExpressionStatement',
+	methodSelector({
+		name: 'push',
+		property: 'expression'
+	})
+].join('');
+
+const selector = `${arrayPushExpressionStatement} + ${arrayPushExpressionStatement}`;
+
+const getCallExpressionArgumentsText = (node, sourceCode) => {
+	const openingParenthesisToken = sourceCode.getTokenAfter(node.callee, isOpeningParenToken);
+	const closingParenthesisToken = sourceCode.getLastToken(node);
+
+	return sourceCode.text.slice(
+		openingParenthesisToken.range[1],
+		closingParenthesisToken.range[0]
+	);
+};
+
+function getFirstExpression(node, sourceCode) {
+	const {parent} = node;
+	const visitorKeys = sourceCode.visitorKeys[parent.type] || Object.keys(parent);
+
+	for (const property of visitorKeys) {
+		const value = parent[property];
+		if (Array.isArray(value)) {
+			const index = value.indexOf(node);
+
+			if (index !== -1) {
+				return value[index - 1];
+			}
+		}
+	}
+
+	/* istanbul ignore next */
+	throw new Error('Can not find the first `Array#push()` call.');
```

Comment:
```suggestion
	throw new Error('Cannot find the first `Array#push()` call.');
```

Reply:
Should it really throw? If it is supposed to catch a case that should not happen, I think we should have some text that tells the user to open an issue.

Reply:
This function is a little bit hacky, I'm not sure if it can cover all cases, I can add a issue link.